design-decisions/2020-05-02-function-rules.md

```diff
@@ -0,0 +1,176 @@
+# Interpretation of function rules
+
+## Purpose
+
+The purpose of this document is to describe in detail how K function rules will be interpreted in Kore.
+We will also summarize the motivation for this decision, that is:
+what problems this decision solves.
+
+## Notation
+
+`Xᵢ` and `Yᵢ` refer to element variables.
+`{X}` and `{Y}` refer to all variables in the respective family of element variables.
+`fun(...)` is a function symbol.
+`φ` are arbitrary patterns appearing as the arguments of a function rule.
+`ψ` are arbitrary patterns appearing on the right-hand side of function rules.
+`Pre` and `Post` are respectively pre- and post-condition patterns.
+
+## Background
+
+The family of K rules defining `fun`,
+
+```.k
+rule fun(φ₁ᵢ({Y}), φ₂ᵢ({Y}), ...) => ψᵢ({Y}) requires Preᵢ({Y}) ensures Postᵢ({Y})
+```
+
+is interpreted in Kore as
```

Comment:
This way of writing it confused me when I first read this document.  I had to come back here to add this comment later, once I understood what was going on.  What I understood first was that this is how function rules are interpreted in Kore and that's it, nothing to do about it.  So I would say that that's how they are interpreted currently, as of May 2020, which is problematic, as explained below.